libraries/AP_NavEKF3/AP_NavEKF3_VehicleStatus.cpp

```diff
@@ -289,6 +289,16 @@ void NavEKF3_core::calcGpsGoodForFlight(void)
     gpsAccuracyGood = gpsSpdAccPass && ekfInnovationsPass;
 }
 
+// Determine if Mag yaw data is good or not
+void NavEKF3_core::calcCompassGoodForFlight()
+{
+    if ((magTestRatio.x > 1.0f || magTestRatio.y > 1.0f || yawTestRatio > 1.0f) && (frontend->_gpsCheck & MASK_GPS_YAW_ERR)) {
```

Comment:
Isn't this inverted? magTestRatio should be < 1 on all axes for the compass to be considered healthy. See
AP_NavEKF3_MagFusion.cpp:563

Reply:
@andyp1per Sorry for the late reply, actually i have taken the reference from AP_NavEKF3_VehicleStatus.cpp:197

Reply:
Right that too indicates that >1 is bad, <1 good - so I stand by my comment